client/rpc_stream.go

```diff
@@ -106,11 +116,15 @@ func (r *rpcStream) Error() error {
 }
 
 func (r *rpcStream) Close() error {
+	r.RLock()
+
 	select {
 	case <-r.closed:
+		r.RUnlock()
 		return nil
 	default:
 		close(r.closed)
+		r.RUnlock()
 		return r.codec.Close()
 	}
 }
```

Comment:
A subsequent issue and patch were added to solve a potential race condition here. See https://github.com/go-micro/go-micro/pull/1643 - we may wish to bring this in too.